### ISSUE TITLE
stack overflow when custom inspect calls Deno.inspect on `this`

### ISSUE THREAD

**Issue by y21** [denoland/deno#9483]
If an object has a custom inspect function attached to it and it calls Deno.inspect with `this` as parameter (so that it has to invoke the same custom inspect function over and over again), Deno **crashes** with a stack overflow error.
Node.js just throws a "Maximum call stack size" error, which can be caught in JavaScript code.
I can only reproduce it on Windows. Running it in WSL throws the expected RangeError. 

Minimal repro:
```js
console.log({
  [Deno.customInspect]: function() {
    return Deno.inspect(this);
  }
});
```

```js
$ deno run index.ts
thread 'main' has overflowed its stack

$ deno --version
deno 1.7.2 (release, x86_64-pc-windows-msvc)
v8 8.9.255.3
typescript 4.1.3
```

**Comment by caspervonb** [comment]
That's a recursive call so, working as intended.

Why would you want to do this?

**Comment by y21** [comment]
Yeah, I would expect some kind of error that can be caught in JS land, which is probably impossible since I guess it's recursive Rust calls causing this. I don't think I've seen a Node.js segfault/crash labeled as "working as intended". This is indeed a bit of a silly repro, but I imagine it could lead to confusion in the future, especially in a larger codebase, and JS developers probably don't want to see/debug stack overflow errors caused by the underlying runtime

**Comment by caspervonb** [comment]
Yeah thats why I wanted clarification if you were raising the issue that inspect on custom inspect itself is recursive as is the title or that it panics.

The panic; we should probably avoid.

**Comment by bnoordhuis** [comment]
What probably happens is that Deno or V8 misdetects the stack size.

@y21 Do you get a regular stack overflow exception with `deno run --v8-flags=--stack_size=512 index.ts`? Try playing around with the size, it's in kilobytes.

**Comment by Jacherr** [comment]
I can reproduce this issue on Windows. Using the default stack size on Windows (1MB) causes Deno to crash, and using stack sizes that are more representative of Linux's default stack size of 8MB causes a runtime error.

**Comment by bnoordhuis** [comment]
I'd be curious to know what the inflection point is for `--stack_size=` where it transitions from "Maximum call stack size" to "thread 'main' has overflowed its stack". The default is 984 but maybe that's too big on some systems.

One important caveat: `--stack_size` doesn't _change_ the stack size, it only tells V8 how big it is.

**Comment by AaronO** [comment]
I guess this is technically 2 separate bugs/issues, as @bnoordhuis pointed out it appears that v8 is misdetecting stack size on some platforms allowing JS to overflow the real stack before exceeding V8's limit and then the `customInspect` circular issue.

@y21 I've submitted a fix (#10338) for the `customInspect` not detecting the circular reference.

@bnoordhuis I would suggest opening a separate issue to track stack size misdetect
